src/atomWithQuery.ts

```diff
@@ -0,0 +1,57 @@
+import type { AnyVariables, Client, Operation } from '@urql/core'
+import { createRequest } from '@urql/core'
+import type { Getter } from 'jotai/vanilla'
+import { clientAtom } from './clientAtom'
+import { createAtoms } from './common'
+import { AtomWithQuery, AtomWithQueryOptions } from './types'
+
+export function atomWithQuery<
+  Data = unknown,
+  Variables extends AnyVariables = AnyVariables
+>(
+  options: AtomWithQueryOptions<Data, Variables>
+): AtomWithQuery<Data, Variables> {
+  const {
+    query,
+    getVariables = () => ({} as Variables),
+    getContext,
+    getClient = (get: Getter) => get(clientAtom),
+    getPause: getPause = () => false,
+  } = options
+  const cache = new WeakMap<Client, Operation>()
+  // This is to avoid recreation of the client on every operation result change
+  // This is to make it more reliable when people do mistakes plus
+  // making client dynamic is rather very bad idea, as all cache is in the client
+  let client: Client
```

Comment:
The same applies to this.

Reply:
Why is `client` recreated on every time?

Reply:
It is not. (aside from examples and tests)

When atomWithQuery gets called this gets closured by all the things below, again, and then gets set in the getArgs function. And then used in functions below (e.g. handleAction).

I don't understand how this variable can be used across providers when it's scoped to simple JS function. I don't see how Jotai can do some magic and collect variable declarations into some catalogue to reuse or so.

Reply:
Changed it so this client and cache weak map is not needed.